src/backend/distributed/metadata/pg_get_object_address.c

```diff
@@ -0,0 +1,503 @@
+/*-------------------------------------------------------------------------
+ *
+ * pg_get_object_address.c
+ *
+ * Copied functions from Postgres pg_get_object_address with acl check.
+ * We need to copy that function to use obtained node to check ownership.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+#include "miscadmin.h"
+
+#include "catalog/objectaddress.h"
+#include "catalog/pg_type.h"
+#include "distributed/citus_ruleutils.h"
+#include "distributed/metadata/distobject.h"
+#include "distributed/pg_version_constants.h"
+#include "utils/array.h"
+#include "utils/builtins.h"
+#include "parser/parse_type.h"
+
+static List * textarray_to_strvaluelist(ArrayType *arr);
+
+/* It is defined on PG >= 13 versions by default */
+#if PG_VERSION_NUM < PG_VERSION_13
+	#define TYPALIGN_INT 'i'
+#endif
+
+/*
+ * Get the object address. If accessCheck is true, access of the user on the object
+ * is checked and if it is not the current user function will error out.
+ *
+ * This function is mostly copied from pg_get_object_address of the PG code. We need
```

Comment:
Maybe mention that `CITUS CODE BEGIN` is the part where citus code begins here as well as this is a really long function and one might not realize easily